src/fsharp/TastOps.fs

```diff
@@ -8569,7 +8570,7 @@ let IsSimpleSyntacticConstantExpr g inputExpr =
         | UncheckedDefaultOfExpr g _ 
         | SizeOfExpr g _ 
         | TypeOfExpr g _ 
-        | NameOfExpr g _ -> true
+        | NameOfExpr g _ when g.langVersion.SupportsFeature LanguageFeature.NameOf -> true
```

Comment:
Ah this is subtle - then `when` actually guards all the different patterns - `sizeof` etc.

I'll submit a change to fix this - this will be the cause of the various test failures